build.c

```diff
@@ -146,6 +146,49 @@ computedirty(struct edge *e, struct node *newest)
 	}
 }
 
+void
+buildupdate(struct node *n)
```

Comment:
I wonder if we could reuse even more code here by passing input and output nodes as parameters to a helper function